packages/mux-player-react/src/index.tsx

```diff
@@ -194,9 +194,27 @@ const usePlayer = (
       return defaultHasChanged(playerEl, value, propName);
     }
   );
-  // NOTE: Somewhere in the codebase, `currentTime` is getting cast to a number, resulting in `NaN` + an error.
-  // This is a bandaid solution for now. (CJP)
-  useObjectPropEffect('currentTime', currentTime ?? 0, ref);
+  // NOTE: Need to handle currentTime differently to manage
+  // 1. Whether the prop value is/was "unset" to e.g. `undefined`
+  //    - To make sure we don't try to update currentTime on these changes
+  // 2. Whether the *prop* value was changed ("from the outside")
+  //    - To only apply the `currentTime` if it actually changed from the last provided value
+  // (CJP)
+  const [internalCurrentTime, setInternalCurrentTime] = useState(currentTime);
+  useEffect(() => {
+    if (internalCurrentTime == currentTime) return;
+    setInternalCurrentTime(currentTime);
+  }, [currentTime, internalCurrentTime]);
```

Comment:
digging a bit more, I believe this code snippet is unneeded because React's effect compare array does the same thing https://github.com/muxinc/elements/blob/741d607521ca9578cfad9f0a9216a6565b4c56a1/packages/mux-player-react/src/useObjectPropEffect.ts#L63-L63

if the previous currentTime was not changed from the current, the effect will not run.

the check `if (currentTimeVal == null) return;` fixes the bug reported here
https://github.com/muxinc/elements/issues/506

does that seem right or am I missing something?

Reply:
Ah so since we only care about the (immediately) previous state as a comparison for the current prop state and since react necessarily stores and compares that "under the hood" for `useEffect` (which we use in `useObjectPropEffect`), that "does all the work needed" for what I'm trying to solve via the `useState` + effect. So between my updated bail + no longer setting the `currentTime = 0` by default, yeah, I believe that should cover it! Brilliant!